File: app/src/main/java/com/example/sun3toline/kepoin/activity/LoginActivity.java
package com.example.sun3toline.kepoin.activity;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextUtils;
import android.view.KeyEvent;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.example.sun3toline.kepoin.R;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.android.material.button.MaterialButton;
import com.google.android.material.textfield.TextInputLayout;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

public class LoginActivity extends AppCompatActivity {
    private EditText mLoginEmailField;
    private EditText mLoginPasswordField;
    private MaterialButton mLoginBtn;
    private FirebaseAuth mAuth;
    private DatabaseReference mDatabaseUser;
    private MaterialButton mRegister;
    private ProgressDialog mProgress;
    private TextInputLayout passwTextInputLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        mAuth = FirebaseAuth.getInstance();
        mProgress = new ProgressDialog((this));
        mRegister = findViewById(R.id.signUpbtn);
        passwTextInputLayout = findViewById(R.id.password_text_input);
        mDatabaseUser = FirebaseDatabase.getInstance().getReference().child("Users");
        mDatabaseUser.keepSynced(true);
        mLoginPasswordField = findViewById(R.id.passfield);
        mLoginPasswordField.setOnKeyListener(new View.OnKeyListener() {
            @Override
            public boolean onKey(View view, int i, KeyEvent keyEvent) {
                if (isPasswordValid(mLoginPasswordField.getText())) {
                    passwTextInputLayout.setError(null); //Clear the error
                }
                return false;
            }
        });
        mLoginEmailField = findViewById(R.id.email1field);
        mLoginBtn = findViewById(R.id.lgnbtn);

        mRegister.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent register = new Intent(LoginActivity.this, RegisterActivity.class);
                register.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
                startActivity(register);
            }
        });

        mLoginBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                checkLogin();
            }
        });


    }

    private void checkLogin() {
        final String email = mLoginEmailField.getText().toString().trim();
        String password = mLoginPasswordField.getText().toString().trim();
        String emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+";

        if (!TextUtils.isEmpty(email) && !TextUtils.isEmpty(password) && email.matches(emailPattern)) {
            mProgress.setMessage("Checking Login...");
            mProgress.show();
            mAuth.signInWithEmailAndPassword(email, password).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                @Override
                public void onComplete(@NonNull Task<AuthResult> task) {

                    if (task.isSuccessful()) {
                        mProgress.dismiss();
                        checkUserExist();

                    } else {
                        mProgress.dismiss();
                        mLoginEmailField.setError("isi dengan lengkap");
                        mLoginPasswordField.setError("teliti lagi password anda");
                        Toast.makeText(LoginActivity.this, "Error Login", Toast.LENGTH_LONG).show();
                    }
                }
            });
        } else {
            mLoginEmailField.setError("isi dengan lengkap");
            mLoginPasswordField.setError("teliti lagi password anda");
        }
    }

    private void checkUserExist() {
        final String user_id = mAuth.getCurrentUser().getUid();
        mDatabaseUser.addValueEventListener(new ValueEventListener() {
            @Override
            public void onDataChange(DataSnapshot dataSnapshot) {
                if (dataSnapshot.hasChild(user_id)) {
                    Intent mainIntent = new Intent(LoginActivity.this, KepoIn.class);
                    mainIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
                    startActivity(mainIntent);
                    finish();
                } else {
                    Intent setupIntent = new Intent(LoginActivity.this, Setup_activity.class);
                    setupIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
                    startActivity(setupIntent);
                    finish();
                }
            }

            @Override
            public void onCancelled(DatabaseError databaseError) {

            }
        });

    }

    private boolean isPasswordValid(@Nullable Editable text) {
        return text != null && text.length() >= 8;
    }
}

File: app/src/main/java/com/achmad/sun3toline/kepoin/activity/LoginActivity.kt
/*
 * *
 *  * Created by Achmad Fathullah on 10/3/20 9:21 PM
 *  * Copyright (c) 2020 . All rights reserved.
 *  * Last modified 10/3/20 8:56 PM
 *
 */

package com.achmad.sun3toline.kepoin.activity

import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextUtils
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.achmad.sun3toline.kepoin.R
import com.achmad.sun3toline.kepoin.utils.dialog.ProgressDialog
import com.google.android.material.button.MaterialButton
import com.google.android.material.textfield.TextInputLayout
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.*

class LoginActivity : AppCompatActivity() {
    private var mLoginEmailField: EditText? = null
    private var mLoginPasswordField: EditText? = null
    private var mLoginBtn: MaterialButton? = null
    private var mAuth: FirebaseAuth? = null
    private var mDatabaseUser: DatabaseReference? = null
    private var mRegister: MaterialButton? = null
    private var mProgress: ProgressDialog? = null
    private var passwTextInputLayout: TextInputLayout? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        mAuth = FirebaseAuth.getInstance()
        mProgress = ProgressDialog(this)
        mRegister = findViewById(R.id.signUpbtn)
        passwTextInputLayout = findViewById(R.id.password_text_input)
        mDatabaseUser = FirebaseDatabase.getInstance().reference.child("Users")
        mDatabaseUser?.keepSynced(true)
        mLoginPasswordField = findViewById(R.id.passfield)
        mLoginPasswordField?.setOnKeyListener { _, _, _ ->
            if (isPasswordValid(mLoginPasswordField?.text)) {
                passwTextInputLayout?.error = null //Clear the error
            }
            false
        }
        mLoginEmailField = findViewById(R.id.email1field)
        mLoginBtn = findViewById(R.id.lgnbtn)
        mRegister?.setOnClickListener {
            val register = Intent(this@LoginActivity, RegisterActivity::class.java)
            register.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
            startActivity(register)
        }
        mLoginBtn?.setOnClickListener { checkLogin() }
    }

    private fun checkLogin() {
        val email = mLoginEmailField?.text.toString().trim { it <= ' ' }
        val password = mLoginPasswordField?.text.toString().trim { it <= ' ' }
        val emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+".toRegex()
        if (!TextUtils.isEmpty(email) && !TextUtils.isEmpty(password) && email.matches(emailPattern)) {
            mProgress?.setMessage("Checking Login...")
            mProgress?.show()
            mAuth?.signInWithEmailAndPassword(email, password)?.addOnCompleteListener { task ->
                if (task.isSuccessful) {
                    mProgress?.dismiss()
                    checkUserExist()
                } else {
                    mProgress?.dismiss()
                    mLoginEmailField?.error = "isi dengan lengkap"
                    mLoginPasswordField?.error = "teliti lagi password anda"
                    Toast.makeText(this@LoginActivity, "Error Login", Toast.LENGTH_LONG).show()
                }
            }
        } else {
            mLoginEmailField?.error = "isi dengan lengkap"
            mLoginPasswordField?.error = "teliti lagi password anda"
        }
    }

    private fun checkUserExist() {
        val userId = mAuth?.currentUser?.uid.toString()
        mDatabaseUser?.addValueEventListener(object : ValueEventListener {
            override fun onDataChange(dataSnapshot: DataSnapshot) {
                if (dataSnapshot.hasChild(userId)) {
                    val mainIntent = Intent(this@LoginActivity, KepoIn::class.java)
                    mainIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK)
                    startActivity(mainIntent)
                    finish()
                } else {
                    val setupIntent = Intent(this@LoginActivity, SetupActivity::class.java)
                    setupIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
                    startActivity(setupIntent)
                    finish()
                }
            }

            override fun onCancelled(databaseError: DatabaseError) {}
        })
    }

    private fun isPasswordValid(text: Editable?): Boolean {
        return text != null && text.length >= 8
    }
}